tests/stub/routing/test_routing_v4x4.py

```diff
@@ -2631,3 +2631,67 @@ def run_and_assert_error(runner):
         self.assertTrue(route_count2 > route_count1 > 0)
         self._writeServer1.done()
         self._readServer1.done()
+
+    def test_should_succeed_when_another_conn_fails_and_discover_using_tx_run(
+            self):
+        # TODO remove this block once fixed
+        if get_driver_name() in ["go"]:
+            self.skipTest("Session close fails with ConnectivityError")
+        # TODO remove this block once fixed
+        if get_driver_name() in ["javascript"]:
+            self.skipTest("Transaction result consumption times out")
+        driver = Driver(self._backend, self._uri_with_context, self._auth,
+                        self._userAgent)
+        self.start_server(
+            self._routingServer1,
+            "router_yielding_writer1_sequentially.script")
+        self.start_server(
+            self._writeServer1,
+            "writer_tx_with_unexpected_interruption_on_run_path.script")
+
+        session1 = driver.session("w", database=self.adb)
+        tx1 = session1.begin_transaction()
+        session2 = driver.session("w", database=self.adb)
+        tx2 = session2.begin_transaction()
```

Comment:
Does the original test also use 2 sessions for this? I'm really curious if 
```python
        session1 = driver.session("w", database=self.adb)
        tx1 = session1.begin_transaction()
        tx2 = session1.begin_transaction()
```

would also work. I'm thinking it should.

Reply:
Yes, it used 2 sessions. Java driver prohibits multiple open transactions.